drivers/misc/nordic_vpr_launcher/nordic_vpr_launcher.c

```diff
@@ -51,19 +51,17 @@ static int nordic_vpr_launcher_init(const struct device *dev)
 	 COND_CODE_0(DT_FIXED_PARTITION_EXISTS(node_id), (0), (DT_REG_ADDR(DT_GPARENT(node_id)))))
 
 #define NORDIC_VPR_LAUNCHER_DEFINE(inst)                                                           \
-	COND_CODE_1(DT_NODE_HAS_PROP(inst, source_memory),                                         \
-		    (BUILD_ASSERT((DT_REG_SIZE(DT_INST_PHANDLE(inst, execution_memory)) ==         \
-				   DT_REG_SIZE(DT_INST_PHANDLE(inst, source_memory))),             \
-				  "Source/execution memory sizes mismatch");),                     \
-		    ())                                                                            \
+	IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, source_memory),                                     \
+		   (BUILD_ASSERT((DT_REG_SIZE(DT_INST_PHANDLE(inst, execution_memory)) <=          \
+				  DT_REG_SIZE(DT_INST_PHANDLE(inst, source_memory))),              \
+				 "Execution memory exceeds source memory size");))                 \
```

Comment:
Shouldn't this be `>=`? The block copied from the source region should fit into the execution one, right?

Reply:
I chose `<=` because PPR's execution memory is 62K by default, while the source memory in MRAM needs to be rounded up to the nearest 4K boundary. The copied block is changed to be the size of the execution memory, so it fits.

Reply:
Oh, I see that not the execution memory size is used in the copying. Sorry for the noise.